indexing-service/src/main/java/org/apache/druid/indexing/common/TaskRealtimeMetricsMonitorBuilder.java

```diff
@@ -41,13 +44,26 @@ public static TaskRealtimeMetricsMonitor build(
     return new TaskRealtimeMetricsMonitor(
         metrics,
         meters,
-        ImmutableMap.of(
-            DruidMetrics.DATASOURCE, new String[]{task.getDataSource()},
-            DruidMetrics.TASK_ID, new String[]{task.getId()},
-            DruidMetrics.TASK_TYPE, new String[]{task.getType()},
-            DruidMetrics.GROUP_ID, new String[]{task.getGroupId()}
-        ),
+        getMetricDimensions(task),
```

Comment:
The `Map<String, String[]> dimensions` on `TaskRealtimeMetricsMonitor` could be changed to `ServiceMetricEvent.Builder`, then it could be passed `task.getMetricBuilder()` and avoid the duplicated logic / instanceof check below.